### PR TITLE
Refs 105: Add pr_checks and build_deploy

### PR DESCRIPTION
**Pull request by mshriver**
Add a docker-login make target, for use in build_deploy.sh

build_deploy.sh will be used by appsre pipelines to push commits to
quay/redhat.io. The GH action building to quay can be removed (once
appinterface updates are accepted and repo created).

pr_checks leverages bonfire for PR image building, deployment, and test execution. appinterface pipeline definitions will execute this script against PRs.

### CODE REVIEW

File: mk/docker.mk
Comment (mshriver) [inline mk/docker.mk:28]: This I loosely tested, seems to work as expected.

File: build_deploy.sh
Comment (mshriver) [inline build_deploy.sh:1]: basically untested at this point, as with pr_checks looking for guidance on cleanly testing.

File: build_deploy.sh
Comment (avisiedo) [inline build_deploy.sh:25]:
```suggestion
make docker-login \
    DOCKER_LOGIN_USER="$RH_REGISTRY_USER" \
    DOCKER_LOGIN_TOKEN="$RH_REGISTRY_TOKEN" \
    DOCKER_REGISTRY="registry.redhat.io"
```
Similar to the above one.

File: mk/docker.mk
Comment (avisiedo) [inline mk/docker.mk:34]:
```suggestion
	$(DOCKER) build $(DOCKER_BUILD_OPTS) -t "$(DOCKER_IMAGE)" -f $(DOCKER_DOCKERFILE) $(DOCKER_CONTEXT_DIR)
```
I'd rather to use a variable, and set that when we need; in this particular scenario, docker will be using cached layers to build the container image except when we pass DOCKER_BUILD_OPTS="--no-cache":

`make docker-build DOCKER_BUILD_OPTS=--no-cache`

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-105

**Comment by jlsherrill** [comment]
⚠️ This task currently requires qe-approval, but this PR does not fully resolve the task.  Please contact QE to determine appropriate testing required.

File: build_deploy.sh
Comment (avisiedo) [inline build_deploy.sh:1]:
```suggestion
#!/bin/bash
set -e
```
I would add shebang even if it does not have executable permissions.

Update `set -exv` by `set -e` before merge to avoid traces about the passwords are printed out in logs. Unless I am missing something about the execution of this script.

File: build_deploy.sh
Comment (avisiedo) [inline build_deploy.sh:4]:
```suggestion
IMAGE_TAG="$(git rev-parse --short=7 HEAD)"
```
I would use quotes `"` to enclose variable value.

File: pr_checks.sh
Comment (avisiedo) [reply pr_checks.sh:1]: no idea about IQE plugin tests honestly, happy to know more about it to see if I can help out.

File: build_deploy.sh
Comment (mshriver) [reply build_deploy.sh:1]: executable permissions are a good point - I'm going to modify perms to make it executable, not sure if the Jenkins jobs are setup to do that automatically. Will just mirror what other service repos set for permissions.

The script is intended to be used by ci-int/ci-ext jenkins instances, ack on updating the set args.

File: pr_checks.sh
Comment (avisiedo) [reply pr_checks.sh:1]: I don't know how to try this script either.

File: build_deploy.sh
Comment (avisiedo) [inline build_deploy.sh:37]:
```suggestion
# Set docker envvars and build
make docker-build docker-push \
    DOCKER_BUILD_OPTS=--no-cache \
    DOCKER_IMAGE_BASE="$IMAGE" \
    DOCKER_IMAGE_TAG="$IMAGE_TAG"
```

File: build_deploy.sh
Comment (avisiedo) [inline build_deploy.sh:39]:
```suggestion
make docker-push \
    DOCKER_IMAGE_BASE="$IMAGE" \
    DOCKER_IMAGE_TAG="$SMOKE_TEST_TAG"
```
I would reuse current rule

File: build_deploy.sh
Comment (avisiedo) [inline build_deploy.sh:38]:
```suggestion
make docker-tag \
    DOCKER_IMAGE_SOURCE="${IMAGE}:${IMAGE_TAG}" \
    DOCKER_IMAGE_TARGET="${IMAGE}:${SMOKE_TEST_TAG}"
```

See below; if the rule `docker-tag` is added, it could look like the above:

File: mk/docker.mk
Comment (avisiedo) [inline mk/docker.mk:32]:
```suggestion

docker-tag: ## Retag DOCKER_IMAGE_SOURCE as DOCKER_IMAGE_TARGET
    [ "$(DOCKER_IMAGE_SOURCE)" != "" ] || { echo "DOCKER_IMAGE_SOURCE ir required"; exit 1; }
    [ "$(DOCKER_IMAGE_TARGET)" != "" ] || { echo "DOCKER_IMAGE_TARGET ir required"; exit 1; }    
    $(DOCKER) tag "$(DOCKER_IMAGE_SOURCE)" "$(DOCKER_IMAGE_TARGET)"

```

File: build_deploy.sh
Comment (avisiedo) [inline build_deploy.sh:4]: For testing this I used:
```suggestion
IMAGE="${IMAGE:-quay.io/cloudservices/content-sources-backend}"
```

And I passed `IMAGE=quay.io/avisied0/content-sources-backend`

Unsure if any impact, into the environment where it will run; if IMAGE is already defined, it will evoke the image generated is not the expected one.

**Comment by jlsherrill** [comment]
Built and pushed image for bb691bd53008b196649a3e6a70112894a2124d67

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/14712517

File: build_deploy.sh
Comment (mshriver) [reply build_deploy.sh:4]: Intended run environment is CI pipelines run by appsre, but I think this change is good to keep it useful outside of that environment without direct modification to the file.

File: mk/docker.mk
Comment (avisiedo) [reply mk/docker.mk:32]: As spoken, for a future pr; resolving this conversation

File: build_deploy.sh
Comment (avisiedo) [reply build_deploy.sh:38]: as spoken, for a future pr; resolving this conversation

File: build_deploy.sh
Comment (mshriver) [reply build_deploy.sh:38]: Thinking this should be part of the docker-tag make rule, or at least addressed in the followup for that rule.

**Review by avisiedo — APPROVED** [review]
LGTM

**Comment by jlsherrill** [comment]
Built and pushed image for 20cf2ba085260631cac10e29d5296a82c1c50d7f

**Review by jlsherrill — APPROVED** [review]

**Pull request merged**
